### PR TITLE
chore: add omitempty to fields

### PR DESCRIPTION
**Pull request by dennisme**
**What this PR does / why we need it**:
My team noticed this when applying some yaml during node creation. Is the v1beta2 stuff used yet? 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by chuckha** [comment]
v1beta2 kubeadm config is not used at this time as it is only supported in >= k8s v1.15

**Comment by rudoi** [comment]
> Is there any workaround you could do here instead of changing the types?

@chuckha Without `omitempty` or the `// +optional` annotation, the generated CRD will mark the field as required in the openAPI spec. Unless there's another way?

**Comment by chuckha** [comment]
that sounds right, I guess I'm confused as to what exactly was seen and why these types should be marked as optional and if it's worth deviating from upstream for this change

**Comment by vincepri** [comment]
Are we planning to copy or use the defaulting logic from k/k in this iteration? If not, we should probably go ahead with this change and mark the field as optional with a comment referencing an issue to remove the defaulting.

We can also consider using admission webhooks to default these fields when they're created.

**Comment by rudoi** [comment]
Yeah as this sits right now, the YAML for a KubeadmConfig gets denied at the front door without a bootstrap token, etc.

**Comment by chuckha** [comment]
ok, got it, thanks for explaining @rudoi 
/ok-to-test
/approve

File: kubeadm/v1beta1/types.go [hunk 1]
```diff
@@ -307,10 +307,10 @@ type JoinConfiguration struct {
 	// CACertPath is the path to the SSL certificate authority used to
 	// secure comunications between node and control-plane.
 	// Defaults to "/etc/kubernetes/pki/ca.crt".
-	CACertPath string `json:"caCertPath"`
+	CACertPath string `json:"caCertPath,omitempty"`
```
[the remark this answers is not in the document]
Comment (dennisme) [reply kubeadm/v1beta1/types.go]: Sure.

File: kubeadm/v1beta1/types.go [hunk 2]
```diff
@@ -307,10 +307,10 @@ type JoinConfiguration struct {
 	// CACertPath is the path to the SSL certificate authority used to
 	// secure comunications between node and control-plane.
 	// Defaults to "/etc/kubernetes/pki/ca.crt".
```
[the remark this answers is not in the document]
Comment (dennisme) [reply kubeadm/v1beta1/types.go]: They both generate the same yaml. I could add it if you want.

[the remark this answers is not in the document]
Comment (vincepri) [reply kubeadm/v1beta1/types.go]: Let's add it for the sake of consistency, it usually goes along side with `omitempty`.

[the remark this answers is not in the document]
Comment (dennisme) [reply kubeadm/v1beta1/types.go]: Deal 🖖

File: kubeadm/v1beta1/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vincepri) [reply kubeadm/v1beta1/types.go]: @dennisme Can we add the same //TODO comment on the other changes? Sorry if I wasn't super clear before!

[the remark this answers is not in the document]
Comment (dennisme) [reply kubeadm/v1beta1/types.go]: Yep

**Pull request merged**